traffic_ops/traffic_ops_golang/iso/iso_test.go

```diff
@@ -0,0 +1,625 @@
+				if got, expectedPrefix := gotResp.Body.String(), mkisofsBin; !strings.HasPrefix(got, expectedPrefix) {
+					t.Errorf("got command: %q; expected command prefix of: %q", got, expectedPrefix)
+				} else {
+					t.Logf("got command: %q", got)
+				}
+			},
+		},
+
+		{
+			name: "cmd failure",
+			input: isoRequest{
+				DHCP:          boolStr{true, false},
+				Stream:        boolStr{true, true},
+				OSVersionDir:  "centos72",
+				HostName:      "db",
+				DomainName:    "infra.ciab.test",
+				IPAddr:        net.IP{172, 20, 0, 4},
+				InterfaceMTU:  1500,
+				InterfaceName: "eth0",
+				IP6Address:    net.IP{},
+				IP6Gateway:    net.IP{},
+				IPGateway:     net.IP{172, 20, 0, 1},
+				IPNetmask:     net.IP{255, 255, 0, 0},
+				MgmtInterface: "",
+				MgmtIPAddress: net.IP{},
+				MgmtIPGateway: net.IP{},
+				MgmtIPNetmask: net.IP{},
+				Disk:          "sda",
+				RootPass:      "12345678",
+			},
+			cmdMod: func(in *exec.Cmd) *exec.Cmd {
+				// Modify command such that it will return an error.
+				return mockISOCmd(in, true, "")
+			},
+			validator: func(t *testing.T, gotResp *httptest.ResponseRecorder) {
+				// TODO: Validate response code. Because of how api.HandleErr works, it's
+				// not possible to see the actual response code in the response recorder.
+
+				// Validate Content-Type header, which should be JSON for this error condition.
+				if got, expected := gotResp.Header().Get(httpHeaderContentType), rfc.ApplicationJSON; got != expected {
+					t.Errorf("header %q = %q; expected: %q", httpHeaderContentType, got, expected)
+				} else {
+					t.Logf("header %q = %q", httpHeaderContentType, got)
+				}
+
+				// Validate that the response body is a JSON-encoded tc.Alerts object.
+				var expectedResp libtc.Alerts
+				if err := json.NewDecoder(gotResp.Body).Decode(&expectedResp); err != nil {
+					t.Fatalf("unable to decode body into expected JSON structure: %v", err)
+				}
+
+				t.Logf("response: %#v", expectedResp)
+			},
+		},
+	}
+
+	tmpDirPrefix := t.Name()
+
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			// Create the OS versions & ks_scripts directories within a temporary directory.
+			// The tmpDir will be used, via a Parameter entry in the database, instead of the default
+			// /var/www/files directory.
+
+			tmpDir, err := ioutil.TempDir("", tmpDirPrefix)
+			if err != nil {
+				t.Fatalf("error creating tempdir: %v", err)
+			}
+			// Clean up temp dir
+			defer os.RemoveAll(tmpDir)
+			// Create expected directory structure
+			if err = os.MkdirAll(filepath.Join(tmpDir, tc.input.OSVersionDir, ksCfgDir), 0777); err != nil {
+				t.Fatal(err)
+			}
+
+			// Create osversions.json file, which is read by validateOSDir method
+			osVersions, err := json.Marshal(libtc.OSVersionsResponse{"test": tc.input.OSVersionDir})
+			if err != nil {
+				t.Fatal(err)
+			}
+			if err = ioutil.WriteFile(filepath.Join(tmpDir, cfgFilename), osVersions, 0777); err != nil {
```

Comment:
I really don't like seeing files made with `0777` permissions; I think this really only needs to be `0644` (or even `0600` might work but maybe a bit overkill).